geode-core/src/main/java/org/apache/geode/cache/query/security/JavaBeanAccessorMethodAuthorizer.java

```diff
@@ -0,0 +1,153 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.cache.query.security;
+
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+
+import org.apache.geode.cache.Cache;
+import org.apache.geode.cache.Declarable;
+
+/**
+ * An immutable and thread-safe {@link MethodInvocationAuthorizer} that allows any method execution
+ * that follows the design patterns for accessor methods described in the JavaBean specification
+ * 1.01, that is, any method whose name begins with "get" or "is". For additional security, only
+ * methods belonging to classes in user-specified packages will be allowed. If a method does not
+ * match the user-specified parameters, then the {@link RestrictedMethodAuthorizer} will make the
+ * final decision
+ *
+ * <p/>
+ *
+ * Some known dangerous methods, like {@link Object#getClass()}, are also rejected by this
+ * authorizer implementation (see
+ * {@link RestrictedMethodAuthorizer#isKnownDangerousMethod(Method, Object)}).
+ * <p/>
+ *
+ * This authorizer implementation addresses only three of the four known security risks:
+ * {@code Java Reflection}, {@code Cache Modification} and {@code Region Modification}.
+ * <p/>
+ *
+ * The {@code Region Entry Modification} security risk still exists: users with the
+ * {@code DATA:READ:RegionName} privilege will be able to execute ANY method (even mutators) on the
+ * objects stored within the region and on instances used as bind parameters of the OQL, so this
+ * authorizer implementation must be used with extreme care.
+ * <p/>
+ *
+ * Usage of this authorizer implementation is only recommended for secured clusters on which only
+ * trusted users and applications have access to the OQL engine. It might also be used on clusters
```

Comment:
The sentence contradicts what we have in the [RFC](https://cwiki.apache.org/confluence/display/GEODE/OQL+Method+Invocation+Security#OQLMethodInvocationSecurity-JavaBeanAccessorMethodAuthorizer). This authorizer is actually intended/recommended to be used on secured clusters on which the operator has fully confidence in that only objects following the _JavaBean specification 1.01_ are stored into regions and used as OQL bind parameters.

Reply:
Should be fixed in the latest commit